nomad/structs/structs.go

```diff
@@ -1157,7 +1157,7 @@ type Service struct {
 // InitFields interpolates values of Job, Task Group and Task in the Service
 // Name. This also generates check names, service id and check ids.
 func (s *Service) InitFields(job string, taskGroup string, task string) {
-	s.Id = GenerateUUID()
+	s.Id = fmt.Sprintf("nomad-%s", GenerateUUID())
```

Comment:
I would create a const.
```
const (
   NomadConsulPrefix = "nomad-"
)
```
And then you can refer to it here: https://github.com/hashicorp/nomad/pull/568/files?diff=unified#diff-2a1fc746b3609034abb284128e9a6ed4R373